config/notifiers.go

```diff
@@ -870,6 +882,77 @@ func (c *JiraConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if c.IssueType == "" {
 		return fmt.Errorf("missing issue_type in jira_config")
 	}
+	return nil
+}
+
+type RocketchatAttachmentField struct {
+	Short *bool  `json:"short"`
+	Title string `json:"title,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type RocketchatAttachmentActionType string
+
+type RocketchatMessageProcessingType string
+
+const (
+	ProcessingTypeSendMessage        RocketchatMessageProcessingType = "sendMessage"
+	ProcessingTypeRespondWithMessage RocketchatMessageProcessingType = "respondWithMessage"
+)
+
+type RocketchatAttachmentAction struct {
+	Type               RocketchatAttachmentActionType  `json:"type,omitempty"`
+	Text               string                          `json:"text,omitempty"`
+	URL                string                          `json:"url,omitempty"`
+	ImageURL           string                          `json:"image_url,omitempty"`
+	IsWebView          bool                            `json:"is_webview"`
+	WebviewHeightRatio string                          `json:"webview_height_ratio,omitempty"`
+	Msg                string                          `json:"msg,omitempty"`
+	MsgInChatWindow    bool                            `json:"msg_in_chat_window"`
+	MsgProcessingType  RocketchatMessageProcessingType `json:"msg_processing_type,omitempty"`
+}
 
+// RocketchatConfig configures notifications via Rocketchat.
+type RocketchatConfig struct {
+	NotifierConfig `yaml:",inline" json:",inline"`
+
+	HTTPConfig *commoncfg.HTTPClientConfig `yaml:"http_config,omitempty" json:"http_config,omitempty"`
+
+	APIURL      *URL    `yaml:"api_url,omitempty" json:"api_url,omitempty"`
+	TokenID     *Secret `yaml:"token_id,omitempty" json:"token_id,omitempty"`
+	TokenIDFile string  `yaml:"token_id_file,omitempty" json:"token_id_file,omitempty"`
+	Token       *Secret `yaml:"token,omitempty" json:"token,omitempty"`
+	TokenFile   string  `yaml:"token_file,omitempty" json:"token_file,omitempty"`
+
+	// RocketChat channel override, (like #other-channel or @username).
+	Channel string `yaml:"channel,omitempty" json:"channel,omitempty"`
+
+	Color       string                        `yaml:"color,omitempty" json:"color,omitempty"`
+	Title       string                        `yaml:"title,omitempty" json:"title,omitempty"`
+	TitleLink   string                        `yaml:"title_link,omitempty" json:"title_link,omitempty"`
+	Text        string                        `yaml:"text,omitempty" json:"text,omitempty"`
+	Fields      []*RocketchatAttachmentField  `yaml:"fields,omitempty" json:"fields,omitempty"`
```

Comment:
I thin we should add a validation rule that limits the maximum number of fields?

Reply:
Hm is that really an issue? Slack fields also don't have such a limit. After all the fields are defined in the configuration not in messages..